### PR TITLE
DENG-3186 Add startup profile reason & attribution experiment to desktop engagement model

### PR DESCRIPTION
**Pull request by kwindau**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3700)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql [hunk 1]
```diff
@@ -12,20 +12,22 @@ SELECT
...
   cfs.attribution_medium,
   cfs.attribution_ua,
+  cfs.attribution_experiment,
   cls.normalized_channel,
   mozfun.norm.os(cls.os) AS normalized_os,
   COALESCE(
     mozfun.norm.windows_version_info(cls.os, cls.os_version, cls.windows_build_number),
     NULLIF(SPLIT(cls.normalized_os_version, ".")[SAFE_OFFSET(0)], "")
   ) AS normalized_os_version,
+  cls.startup_profile_selection_reason_first,
```
Comment (bani) [inline sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql:22]: I'm thinking about renaming this to startup_profile_selection_reason, which is the name in some other tables.
My understanding is that in theory this shouldn't change.

File: sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql [hunk 2]
```diff
@@ -12,20 +12,22 @@ SELECT
   cfs.attribution_dlsource,
   cfs.attribution_medium,
   cfs.attribution_ua,
+  cfs.attribution_experiment,
```
Comment (bani) [inline sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql:15]: we need both `attribution_experiment` and `attribution_variation`. Which I know we don't have on first_seen but we should have on last_seen.

File: sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql (hunk 1 above)
Comment (kwindau) [reply sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql:22]: I'm for it, it's nicer shorter anyway

File: sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql (hunk 2 above)
Comment (kwindau) [reply sql/moz-fx-data-shared-prod/telemetry_derived/desktop_engagement_clients_v1/query.sql:15]: done

**Review by bani — APPROVED** [review]
Thank you!

**Pull request merged**